charms/kfp-profile-controller/src/components/pebble_components.py

```diff
@@ -0,0 +1,65 @@
+import dataclasses
+import logging
+
+from charmed_kubeflow_chisme.components.pebble_component import PebbleServiceComponent
+from ops.pebble import Layer
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class KfpProfileControllerInputs:
+    """Defines the required inputs for KfpProfileControllerPebbleService."""
+
+    MINIO_SECRET: str
+    MINIO_HOST: str
+    MINIO_NAMESPACE: str
+    MINIO_PORT: str
+    MINIO_ACCESS_KEY: str
+    MINIO_SECRET_KEY: str
+    KFP_VERSION: str
+    KFP_DEFAULT_PIPELINE_ROOT: str
+    DISABLE_ISTIO_SIDECAR: str
+    CONTROLLER_PORT: int
+    METADATA_GRPC_SERVICE_HOST: str
+    METADATA_GRPC_SERVICE_PORT: str
+
+
+class KfpProfileControllerPebbleService(PebbleServiceComponent):
```

Comment:
@i-chvets added some nice validation in his PR - check out the discussion [here](https://github.com/canonical/kfp-operators/pull/268/files#r1282361597)